### PR TITLE
Replaces bold HTML tag with ++ in certain secbot/mulebot messages

### LINKED ISSUE #84130: Beepsky has broken text
Reporting client version: 515.1640

## Round ID:
[230775](https://scrubby.melonmesa.com/round/230775)

## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)
- [Bitrunning: Tweaks, QoL and removals](https://www.github.com/tgstation/tgstation/pull/84125)
- [Adds a new power storage type: The Megacell. Drastically reduces power cell consumption/storage. [MDB Ignore]](https://www.github.com/tgstation/tgstation/pull/84079)

## Reproduction:
image

### PR DESCRIPTION
**Pull request by combustor**
## About The Pull Request

Adds RUNECHAT_BOLD which just emboldens messages and uses it instead of `<b></b>` in secbots and mulebots

## Why It's Good For The Game
Secbots and mulebots actually show the `<b></b>` tag in chat rather than making the text bold, replacing it with ++ emboldens it properly.
340880630-de5dc932-3e75-4713-8609-d29f13a0adc3

Fixes #84130

## Changelog

:cl:  Guestify
fix: Fixed secbots and mulebots bold text
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
this isn't really intuitive, i think we should make a `#define` macro helpers in the file where all the text helpers are and have it be named `RUNECHAT_BOLD()` or something. it would just look like

```dm
#define RUNECHAT_BOLD(str) "+[str]+"

var/string = "The last word of this sentence is [RUNECHAT_BOLD("bold")]." // resolves to "The last word of this sentence is +bold+."
```

This is just so people in the future will know why we have plus's on either side, and will allow for better auditing in case someone breaks this method in the future.

**Comment by combustor** [comment]
> this isn't really intuitive, i think we should make a `#define` macro helpers in the file where all the text helpers are and have it be named `RUNECHAT_BOLD()` or something.

Is it okay now?

**Review by Cyberboss — APPROVED** [review]
Assuming you got all the locations

**Review by Cyberboss — CHANGES_REQUESTED** [review]
>`ERROR: File(s) with no trailing newline detected, please add one.` chat.dm

**Review by Cyberboss — APPROVED** [review]

**Comment by carlarctg** [comment]
are there other macros like that? if we have one we should have all of em aswell

**Comment by san7890** [comment]
> are there other macros like that? if we have one we should have all of em aswell

ideally we would have a task-specific PR to add these macros but let's just make do with what we have

File: code/__HELPERS/chat.dm [hunk 1]
```diff
@@ -97,3 +97,6 @@ it will be sent to all connected chats.
 	for(var/mob/creature as anything in mob_list)
 		to_chat(creature, message)
 	send_to_observers(message, source)
+
+/// Emboldens runechat messages
+#define RUNECHAT_BOLD(str) "+[str]+"
```
Comment (san7890) [inline code/__HELPERS/chat.dm:102]: global defines should go in `__DEFINES`, not `__HELPERS`. there should be a big list of `span_x()` macros somewhere in there, that file might work? there's probably another chat.dm there though

**Comment by combustor** [comment]
> are there other macros like that? 

I have no idea to be honest

File: code/__HELPERS/chat.dm (hunk 1 above)
Comment (combustor) [reply code/__HELPERS/chat.dm]: Ah alright, thanks
I've moved it to __DEFINES\chat.dm if that's okay

**Comment by san7890** [comment]
> > are there other macros like that?
> 
> I have no idea to be honest

i think they meant to ask "are there other text modifiers like that that we need to modify", like for `_` and `%` wraps, but that can (and should) be done later since that is a whole other thing to start auditing and figuring out.

**Pull request merged**